WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewModel.swift

```diff
@@ -730,10 +733,40 @@ private extension ProductFormViewModel {
 
     /// Performs initial fetch from storage and updates results.
     func configureResultsController() {
+        blazeCampaignResultsController.onDidChangeContent = { [weak self] in
+            self?.updateBlazeCampaignResult()
+
+        }
+        blazeCampaignResultsController.onDidResetContent = { [weak self] in
+            self?.updateBlazeCampaignResult()
+        }
+
         do {
             try blazeCampaignResultsController.performFetch()
+            updateBlazeCampaignResult()
         } catch {
             ServiceLocator.crashLogging.logError(error)
         }
     }
+
+    func updateBlazeCampaignResult() {
+        hasActiveBlazeCampaign = hasBlazeCampaign()
+    }
+}
+
+// MARK: Blaze
+//
+private extension ProductFormViewModel {
+    /// Check whether there is already an existing campaign for the current Product, that also has one of these statuses:
+    /// - created (in moderation),
+    /// - approved,
+    /// - scheduled, or
+    /// - active.
+    func hasBlazeCampaign() -> Bool {
+        let campaigns = blazeCampaignResultsController.fetchedObjects
+        return campaigns.contains(where: {
+            ($0.productID == product.productID) &&
+            ($0.status == .created || $0.status == .approved || $0.status == .scheduled || $0.status == .active)
```

Comment:
❓ Are we aligned with Android about these statuses?

Reply:
Yup, just happened to coordinate about this with @JorgeMucientes yesterday during pairing. Android:

https://github.com/woocommerce/woocommerce-android/pull/10079/files#diff-0fc9356c23ebbaf4e2b7a41dc809e0b9ae737934830fa67fc8a61521a9818574R18-R20

The difference is that Android doesn't have `approved` status but iOS does. The rest is the same, though, and I think adding `approved` makes sense.

Reply:
Just curious are there any docs on all the possible statuses a campaign has? When working on the feature, I basically copied the statuses from Jetpack Android app code: 

```kotlin
                "created" -> InModeration
                "scheduled" -> Scheduled
                "active" -> Active
                "finished" -> Completed
                "rejected" -> Rejected
                "canceled" -> Canceled
```
Do you know when does `approve` status happens? I though when a campaign was approved it would transition directly to either `active` or `scheduled`.

Reply:
We followed [WPKit iOS mapping](https://github.com/wordpress-mobile/WordPressKit-iOS/blob/1ed1cddc0d17d79be190a201ae645c525b874eb4/WordPressKit/BlazeCampaign.swift#L47-L56) and `approved` is one of the status cases. I'm not sure why there was a difference between the two platforms, but my guess is this case is internal to the server and might not be returned through the API, so it was skipped on Android?

Reply:
I've asked in Slack convo to see if we can get extra context and find out if we have to add the missing values in Android codebases :) 
p1699865671745509-slack-C04LJFS1G5P